src/main/java/appeng/menu/implementations/MenuTypeBuilder.java

```diff
@@ -263,19 +261,12 @@ private I getHostFromPlayerInventory(Player player, MenuLocator locator) {
         if (it.getItem() instanceof IGuiItem guiItem) {
             // Optionally contains the block the item was used on to open the menu
             BlockPos blockPos = locator.hasBlockPos() ? locator.getBlockPos() : null;
-            IGuiItemObject guiObject = guiItem.getGuiObject(it, locator.getItemIndex(), player.level, blockPos);
+            IGuiItemObject guiObject = guiItem.getGuiObject(it, locator.getItemIndex(), player, blockPos);
             if (hostInterface.isInstance(guiObject)) {
                 return hostInterface.cast(guiObject);
             }
         }
 
-        if (hostInterface.isAssignableFrom(WirelessTerminalGuiObject.class)) {
```

Comment:
Wait, if you remove this here (which I think is good), why did you keep the actual `WirelessTerminals` class? :D